src/Options.php

```diff
@@ -9,29 +9,67 @@ class Options
      *
      * @var array
      */
-    public $fileTypes = [".php"];
+    private $fileTypes = [".php"];
 
     /**
-     * @param \stdClass|null $options
+     * @param \Traversable|\stdClass|array|null $options
      */
-    public function __construct(\stdClass $options = null)
+    public function __construct($options = null)
     {
         // Do nothing when the $options parameter is not an object
-        if (!is_object($options)) {
+        if (!is_object($options) && !is_array($options) && (!$options instanceof \Traversable)) {
             return;
         }
 
-        $this->fileTypes = $options->fileTypes ?? $this->normalizeFileTypes($this->fileTypes);
+        foreach ($options as $option => $value) {
+            $method = 'set' . ucfirst($option);
+
+            call_user_func([$this, $method], $value);
```

Comment:
Will it notify the client about the invalid value type?
What about arrays that could have any values as type? E.g. `$fileTypes = ['.php', false]`

Since the options are coming from the client we should validate/filter them